src/server/unit_sao.cpp

```diff
@@ -75,20 +75,19 @@ void UnitSAO::setAnimationSpeed(float frame_speed)
 	m_animation_speed_sent = false;
 }
 
-void UnitSAO::setBonePosition(const std::string &bone, v3f position, v3f rotation)
+void UnitSAO::setBoneOverride(const std::string &bone, BoneOverride *props)
 {
 	// store these so they can be updated to clients
-	m_bone_position[bone] = core::vector2d<v3f>(position, rotation);
-	m_bone_position_sent = false;
+	auto prev_props = m_bone_override[bone];
+	if (prev_props)
+		delete prev_props;
+	m_bone_override[bone] = props;
+	m_bone_override_sent = false;
 }
 
-void UnitSAO::getBonePosition(const std::string &bone, v3f *position, v3f *rotation)
+BoneOverride *UnitSAO::getBoneOverride(const std::string &bone)
 {
-	auto it = m_bone_position.find(bone);
-	if (it != m_bone_position.end()) {
-		*position = it->second.X;
-		*rotation = it->second.Y;
-	}
+	return m_bone_override[bone];
```

Comment:
```suggestion
	auto it = m_bone_override.find(bone);
	return it != m_bone_override.end() ? it->second : nullptr;
```
.. because indexing creates a new entry which then shows up in `player:get_bone_overrides()`.